### ISSUE TITLE
Default colors for Labels

### ISSUE THREAD

**Issue by thoechsmann** [getredash/redash#2873]
We have lots of graphs where we have the same groups. We always have to assign colors manually so the same label gets the same color. 

For example we have many plots grouped by the platform like Android, iOS, Facebook. It would be great if we could maybe globally assign default colors to a label name.

**Comment by kravets-levko** [comment]
Hi! You can manually set colors for charts on the "Colors" settings tab:
![image](https://user-images.githubusercontent.com/12139186/46206834-fff52280-c32d-11e8-930e-7a8c54bde16f.png)

**Comment by thoechsmann** [comment]
Yes, I know. And we do that. But we have LOTS of charts with the same labels and we have to set the color for all of them in every single query and even in every single visualisation.

**Comment by arikfr** [comment]
I would love to find some solution that doesn't require creating configuration for colors, as this introduces its own challenges. If anyone got ideas, don't be shy 🤓

**Comment by jezdez** [comment]
Maybe we could have an "autocolor" mode that is enabled with a single boolean setting, that would automatically populate the color values in a predictable way, e.g. alternating complementary colors that vary by a few grades of R, G or B per iteration? Maybe start at a not-quite-white and end at a not-quite-black?

**Comment by arikfr** [comment]
@jezdez how is it different than today's auto color picker?

**Comment by jezdez** [comment]
Oh right, I missed the important part 😬: "..automatically populate the color values in a predictable way based on the label".

**Comment by WesleyBatista** [comment]
_just brainstorming here_

could be a `column_name: color` mapping on the dashboard level that is applied to every widget on the dashboard when you access it. If you use the "Colors" settings tab from the query visualization, it should take priority.

**Comment by WesleyBatista** [comment]
I did not stopped to see how the code is nowadays, but ideally I imagine the settings being like:

"redash global settings" > "dashboard settings" > "query settings" > "query visualization settings"

**Comment by deecay** [comment]
Maybe 'autocolor' will take series name such as 'male' or 'female', and takes md5hash of it, then take lowest byte of the hash to calculate red, second to lowest for blue, etc. Calculation could be bit-AND, or mod 256 or whatever. This should always assign a fixed color to any aeries by default.

**Comment by arikfr** [comment]
@deecay and in case of collision, it will pick the next available color?

**Comment by deecay** [comment]
@arikfr no I was thinking of one way only. Md5 or other hasing function might give us just enogh randomness.

**Comment by kravets-levko** [comment]
@arikfr @deecay `d3` has [ordinal color scales](https://d3-wiki.readthedocs.io/zh_CN/master/Ordinal-Scales/) intended exactly for this usage pattern (and no `md5` needed)

**Comment by arikfr** [comment]
If I'm not mistaken, d3 scales function will return the same color for the
same input value, but only for the current invocation /instance. But you
might get different color between page refreshes.

I might be wrong, so worth checking.

**Comment by arikfr** [comment]
Might be relevant/useful:
https://gist.github.com/mikhailov-work/ee72ba4191942acecc03fe6da94fc73f
https://observablehq.com/@mbostock/turbo
